src/humanize.js

```diff
@@ -0,0 +1,205 @@
+  },
+  SI: {
+    units: ["", "k", "M", "G", "T", "P", "E"],
+    space: false,
+    divisor: 1000,
+  },
+  decimalBytesPerSec: {
+    units: ["Bps", "KBps", "MBps", "GBps", "TBps", "PBps", "EBps"],
+    space: true,
+    divisor: 1000,
+  },
+  packetsPerSec: {
+    units: ["pps", "kpps"],
+    space: true,
+    divisor: 1000,
+  },
+  seconds: {
+    units: ["ns", "μs", "ms", "s"],
+    space: true,
+    divisor: 1000,
+  },
+};
+
+export const getType = (name) => {
+  const type = TYPES[name];
+  if (!_.isPlainObject(type)) {
+    return {
+      units: [],
+      space: false,
+      divisor: 1000,
+    };
+  }
+  return type;
+};
+
+const convertBaseValueToUnits = (
+value,
+unitArray,
+divisor,
+initialUnit,
+preferredUnit
+) => {
+const sliceIndex = initialUnit ? unitArray.indexOf(initialUnit) : 0;
+const units_ = unitArray.slice(sliceIndex);
+
+if (preferredUnit || preferredUnit === "") {
+  const unitIndex = units_.indexOf(preferredUnit);
+  if (unitIndex !== -1) {
+    return {
+      value: value / divisor ** unitIndex,
+      unit: preferredUnit,
+    };
+  }
+}
+
+let unit = units_.shift();
+while (value >= divisor && units_.length > 0) {
+  value = value / divisor;
+  unit = units_.shift();
+}
+return { value, unit };
+};
+
+const getDefaultFractionDigits = (value) => {
+if (value < 1) {
+  return 3;
+}
+if (value < 100) {
+  return 2;
+}
+return 1;
+};
+
+const formatValue = (value, options) => {
+const fractionDigits = getDefaultFractionDigits(value);
+const { locales, ...rest } = _.defaults(options, {
+  maximumFractionDigits: fractionDigits,
+});
+
+// 2nd check converts -0 to 0.
+if (!isFinite(value) || value === 0) {
+  value = 0;
+}
+return Intl.NumberFormat(locales, rest).format(value);
+};
+
+const round = (units.round = (value, fractionDigits) => {
+if (!isFinite(value)) {
+  return 0;
+}
+const multiplier = Math.pow(
+  10,
+  fractionDigits || getDefaultFractionDigits(value)
+);
+return Math.round(value * multiplier) / multiplier;
+});
+
+
+const humanize = (units.humanize = (
+value,
+typeName,
+useRound = false,
+initialUnit,
+preferredUnit
+) => {
+const type = getType(typeName);
+
+if (!isFinite(value)) {
+  value = 0;
+}
+
+let converted = convertBaseValueToUnits(
+  value,
+  type.units,
+  type.divisor,
+  initialUnit,
+  preferredUnit
+);
+
+if (useRound) {
+  converted.value = round(converted.value);
+  converted = convertBaseValueToUnits(
+    converted.value,
+    type.units,
+    type.divisor,
+    converted.unit,
+    preferredUnit
+  );
+}
+
+const formattedValue = formatValue(converted.value);
+
+return {
+  string: type.space
+    ? `${formattedValue} ${converted.unit}`
+    : formattedValue + converted.unit,
+  unit: converted.unit,
+  value: converted.value,
+};
+});
+
+export const humanizeBinaryBytesWithoutB = (v, initialUnit, preferredUnit) =>
+  humanize(v, "binaryBytesWithoutB", true, initialUnit, preferredUnit);
+export const humanizeBinaryBytes = (v, initialUnit, preferredUnit) =>
```

Comment:
As a standard we use `humanizeBinaryBytes` across console FYI.
Also add components which you will be needing for this PR, rest can be added whenever required. It would allow not to have unnecessary code.